_static/link_gen/link.js

```diff
@@ -141,28 +154,58 @@ function changeTab(div) {
  * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
  */
 function generateCloneDirectoryName(gitCloneUrl) {
+    if(gitCloneUrl.slice(-1) == "/")
+        gitCloneUrl = gitCloneUrl.slice(0,-1);
     var lastPart = gitCloneUrl.split('/').slice(-1)[0];
     return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
 }
 
+function handleSource(args){
```

Comment:
Since this returns structured data from incoming structured data, I think the name `handleSource` isn't very clear. Could you make it more explicit and write a short docstring?

I understand from code inspection that this takes some kind of `args`, which I guess relates to what has been passed via a form submission or similar, and then that it parses those into another structure.

Hmmm, perhaps "parse" would make sense to include in the function name, maybe "form data", maybe also "into ..."? For a helper function that is only used internally etc, having a long descriptive name is something I find very helpful.

Could you write a sentence or two about this as a docstring, and then tweak the name of the function?

Reply:
I renamed this to : configureContentProviderAttrs and added some documentation